### ISSUE TITLE
Adding LeaseID checking in Txn needs a Cmp type method

### ISSUE THREAD

**Issue by purpleidea** [coreos/etcd#8487]
I'd like to do Txn's like this:

```
If v3.Compare(v3.LeaseValue(keyPath), "=", someLeaseID) ...
```

Thanks to some lovely work @heyitsanthony pointed me to, you could write:

```
	LeaseValue := func(key string) v3.Cmp {
		return v3.Cmp{Key: []byte(key), Target: pb.Compare_LEASE}
	}
```

I'd propose adding this as a helper function alongside https://godoc.org/github.com/coreos/etcd/clientv3#Value but there are apparently some naming convention issues in "mapping directly to the mvccpb.KeyValue".

These are beyond my understanding of the internals, but this issue is to track this so interested users get a notification when this is fixed so we can update our code.

Thanks for the great feature!

**Comment by purpleidea** [comment]
I wrote a patch adding this if anyone would like it.

https://github.com/coreos/etcd/pull/8488

**Comment by davissp14** [comment]
https://github.com/coreos/etcd/blob/master/etcdserver/etcdserverpb/rpc.proto#L520

This feature doesn't seem to be working as expected.  I am working on adding support for this here: https://github.com/davissp14/etcdv3-ruby/pull/95/files 

The txn lease comparison seems to ALWAYS return successfully regardless of the comparison.  Is this issue relevant here?  If not, let me know and I can open up a new issue.

**Comment by xiang90** [comment]
@davissp14 what is the server version? the feature only works on master.

**Comment by davissp14** [comment]
@xiang90  That would explain it.  The server version I am testing against is 3.2.6.

**Comment by heyitsanthony** [comment]
Fixed in master. Closing.

**Comment by danevandyck** [comment]
Have you considered adding comparison on lease ID to 3.2?

**Comment by gyuho** [comment]
@danevandyck This was fixed via https://github.com/coreos/etcd/pull/8324.
And will be included in our upcoming 3.3 release.
